samples/snippets/src/main/java/com/example/spanner/BatchWriteAtLeastOnceSample.java

```diff
@@ -0,0 +1,133 @@
+              .set("SingerId")
+              .to(16)
+              .set("FirstName")
+              .to("Scarlet")
+              .set("LastName")
+              .to("Terry")
+              .build());
+  private static final MutationGroup MUTATION_GROUP2 =
+      MutationGroup.of(
+          Mutation.newInsertOrUpdateBuilder("Singers")
+              .set("SingerId")
+              .to(17)
+              .set("FirstName")
+              .to("Marc")
+              .build(),
+          Mutation.newInsertOrUpdateBuilder("Singers")
+              .set("SingerId")
+              .to(18)
+              .set("FirstName")
+              .to("Catalina")
+              .set("LastName")
+              .to("Smith")
+              .build(),
+          Mutation.newInsertOrUpdateBuilder("Albums")
+              .set("SingerId")
+              .to(17)
+              .set("AlbumId")
+              .to(1)
+              .set("AlbumTitle")
+              .to("Total Junk")
+              .build(),
+          Mutation.newInsertOrUpdateBuilder("Albums")
+              .set("SingerId")
+              .to(18)
+              .set("AlbumId")
+              .to(2)
+              .set("AlbumTitle")
+              .to("Go, Go, Go")
+              .build());
+
+  static void batchWriteAtLeastOnce() {
+    // TODO(developer): Replace these variables before running the sample.
+    final String projectId = "my-project";
+    final String instanceId = "my-instance";
+    final String databaseId = "my-database";
+    batchWriteAtLeastOnce(projectId, instanceId, databaseId);
+  }
+
+  static void batchWriteAtLeastOnce(String projectId, String instanceId, String databaseId) {
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseId dbId = DatabaseId.of(projectId, instanceId, databaseId);
+      final DatabaseClient dbClient = spanner.getDatabaseClient(dbId);
+
+      // Creates and issues a BatchWrite RPC request that will apply the mutation groups
+      // non-atomically and respond back with a stream of BatchWriteResponse.
+      ServerStream<BatchWriteResponse> responses =
+          dbClient.batchWriteAtLeastOnce(
+              ImmutableList.of(MUTATION_GROUP1, MUTATION_GROUP2),
+              Options.tag("batch-write-tag"));
+
+      // Iterates through the results in the stream response and prints the MutationGroup indexes,
+      // commit timestamp and status.
+      for (BatchWriteResponse response : responses) {
+        if (response.getStatus().getCode() == Code.OK_VALUE) {
+          System.out.printf(
+              "Mutation group indexes %s have been applied with commit timestamp %s",
+              response.getIndexesList(), response.getCommitTimestamp());
+        } else {
+          System.out.printf(
+              "Mutation group indexes %s could not be applied with error code %s",
+              response.getIndexesList(), Code.forNumber(response.getStatus().getCode()));
```

Comment:
nit: Is it possible to also get the error message and print that? Getting for example an `INVALID_ARGUMENT` error code without any further details will make it hard to debug the error.